starlark/src/stdlib/macros/mod.rs

```diff
@@ -140,12 +153,17 @@ macro_rules! starlark_fun {
     ($(#[$attr:meta])* $fn:ident ( $($signature:tt)* ) { $($content:tt)* } $($($rest:tt)+)?) => {
         $(#[$attr])*
         fn $fn(
-            __call_stack: &mut $crate::eval::call_stack::CallStack,
-            __env: $crate::environment::TypeValues,
-            mut args: $crate::values::function::ParameterParser,
+            call_stack: &mut $crate::eval::call_stack::CallStack,
+            env: $crate::environment::TypeValues,
```

Comment:
why does the signature of this method change? I don't understand why args was mutable so I can see that mut going away but why `call_stack` or `env` should be made accessible to the content?

Reply:
> why args was mutable

It is passed by value, mutable or immutable is the same.

> why call_stack or env should be made accessible to the content

`call_stack` need to be accessible to be passed further down the stack (e. g. in implementation of `sorted` function).

`env` is needed in the implementation of `dir` builtin.